runtime/compiler/runtime/JITaaSIProfiler.cpp

```diff
@@ -294,8 +294,11 @@ TR_JITaaSIProfiler::profilingSample(TR_OpaqueMethodBlock *method, uint32_t byteC
             if (storage->ID == TR_IPBCD_CALL_GRAPH)
                {
                U_8* pc = (U_8*)entry->getPC();
-               if ((*pc == JBinvokeinterface2) && (*(pc + 2) == JBinvokeinterface)) // FIXME: must test the length of the method
+               if ((*pc == JBinvokeinterface2) && (*(pc + 2) == JBinvokeinterface))
+                  {
                   bci += 2;
+                  TR_ASSERT(bci < TR::Compiler->mtd.bytecodeSize(method), "Bytecode index can't be higher than the methodSize: bci=%u methodSize=%u", bci, TR::Compiler->mtd.bytecodeSize(method));
```

Comment:
The ASSERT needs to be put just before dereferencing `pc`. The code will become uglier because you need two separate asserts for `pc` and `pc+2`

Reply:
Do you mean assert for `bci` and `bci + 2`

Reply:
yes, any time we dereference `pc`